gax-httpjson/src/main/java/com/google/api/gax/httpjson/ManagedHttpJsonChannel.java

```diff
@@ -43,62 +46,139 @@
 import com.google.api.core.BetaApi;
 import com.google.api.core.SettableApiFuture;
 import com.google.api.gax.core.BackgroundResource;
+import com.google.api.pathtemplate.PathTemplate;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.base.Strings;
 import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.security.GeneralSecurityException;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.Executor;
 import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
 
 @BetaApi
 public class ManagedHttpJsonChannel implements HttpJsonChannel, BackgroundResource {
   private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
 
   private final Executor executor;
-  private final HttpTransport httpTransport;
   private final String endpoint;
   private final JsonFactory jsonFactory;
   private final ImmutableList<HttpJsonHeaderEnhancer> headerEnhancers;
+  private final GoogleCredentials googleCredentials;
+  private final HttpTransport httpTransport;
+  private final HttpRequestFactory requestFactory;
+
+  private boolean isTransportShutdown;
 
   private ManagedHttpJsonChannel(
       Executor executor,
-      HttpTransport httpTransport,
       String endpoint,
       JsonFactory jsonFactory,
-      List<HttpJsonHeaderEnhancer> headerEnhancers) {
+      List<HttpJsonHeaderEnhancer> headerEnhancers,
+      @Nullable GoogleCredentials googleCredentials) {
     this.executor = executor;
-    this.httpTransport = httpTransport;
     this.endpoint = endpoint;
     this.jsonFactory = jsonFactory;
     this.headerEnhancers = ImmutableList.copyOf(headerEnhancers);
+
+    HttpRequestFactory requestFactory;
+    HttpTransport httpTransport;
+    GoogleCredentials defaultCredentials = null;
+    if (googleCredentials == null) {
+      try {
+        defaultCredentials = GoogleCredentials.getApplicationDefault();
```

Comment:
The  credentials need to come from the CredentialsProvider on ClientSettings

Reply:
Done.